pkg/e2e/operators/ocmagent.go

```diff
@@ -2,218 +2,225 @@ package operators
-				}).Namespace(operatorNamespace).Get(ctx, ocmAgentServiceMonitorName, metav1.GetOptions{})
-				if err != nil {
-					return false, fmt.Errorf("unable to retrieve ocm-agent-metrics servicemonitor")
-				}
+	ginkgo.It("cluster roles exist", label.Install, func(ctx context.Context) {
+		clusterRoles, err := h.Kube().RbacV1().ClusterRoles().List(ctx, metav1.ListOptions{})
+		expect.NoError(err)
+		found := false
+		for _, clusterRole := range clusterRoles.Items {
+			if strings.HasPrefix(clusterRole.Name, clusterRolePrefix) {
+				found = true
+			}
+		}
+		Expect(found).To(BeTrue())
+	})
 
-				if obj.GetName() != ocmAgentServiceMonitorName {
-					return false, fmt.Errorf("ocm-agent-metrics servicemonitor not found")
-				}
-				return true, nil
-			})
-			Expect(err).NotTo(HaveOccurred())
-		}, pollingDuration.Seconds())
-
-		util.GinkgoIt("ocm-agent service should be restored when it gets deleted", func(ctx context.Context) {
-			err := deleteService(ctx, ocmAgentServiceName, operatorNamespace, h)
-			Expect(err).NotTo(HaveOccurred())
-
-			err = wait.Poll(30*time.Second, 5*time.Minute, func() (bool, error) {
-				obj, err := h.Dynamic().Resource(schema.GroupVersionResource{
-					Group: "", Version: "v1", Resource: "services",
-				}).Namespace(operatorNamespace).Get(ctx, ocmAgentServiceName, metav1.GetOptions{})
-				if err != nil {
-					return false, fmt.Errorf("unable to retrieve ocm-agent service")
-				}
+	ginkgo.It("cluster role bindings exist", label.Install, func(ctx context.Context) {
+		clusterRoleBindings, err := h.Kube().RbacV1().ClusterRoleBindings().List(ctx, metav1.ListOptions{})
+		expect.NoError(err)
+		found := false
+		for _, clusterRoleBinding := range clusterRoleBindings.Items {
+			if strings.HasPrefix(clusterRoleBinding.Name, clusterRolePrefix) {
+				found = true
+			}
+		}
+		Expect(found).To(BeTrue())
+	})
 
-				if obj.GetName() != ocmAgentServiceName {
-					return false, fmt.Errorf("ocm-agent service not found")
-				}
-				return true, nil
-			})
-			Expect(err).NotTo(HaveOccurred())
-		}, pollingDuration.Seconds())
-
-		util.GinkgoIt("ocm-agent-allow-only-alertmanager  networkpolicy should restored when it gets deleted", func(ctx context.Context) {
-			err := deleteNetworkPolicy(ctx, ocmAgentNetworkpolicyName, operatorNamespace, h)
-			Expect(err).NotTo(HaveOccurred())
-
-			err = wait.Poll(30*time.Second, 5*time.Minute, func() (bool, error) {
-				obj, err := h.Dynamic().Resource(schema.GroupVersionResource{
-					Group: "networking.k8s.io", Version: "v1", Resource: "networkpolicies",
-				}).Namespace(operatorNamespace).Get(ctx, ocmAgentNetworkpolicyName, metav1.GetOptions{})
-				if err != nil {
-					return false, fmt.Errorf("unable to retrieve ocm-agent-allow-only-alertmanager  networkpolicy")
-				}
+	ginkgo.It("can be upgraded from previous version", label.Upgrade, func(ctx context.Context) {
+		performUpgrade(ctx, h, namespace, operatorName, operatorName, operatorRegistry, 5, 30)
+	})
```

Comment:
since these tests are ordered, can we move this to be the last test?

Reply:
Done!